packages/cli-dxns/src/handlers/deploy/register.ts

```diff
@@ -42,52 +42,45 @@ export const register = ({ getDXNSClient, module, cid, license, account }: Regis
   const record = {
     license,
     ...dataRecord,
-    ...clean({ version }),
-    ...clean({ tag })
+    // TODO(wittjosiah): Include version in generic way?
+    ...clean({ build: { tag } })
   };
 
-  if (record.version === 'false') {
-    record.version = null;
-  }
-
   // Inject IPFS CID.
   set(record, hashPath, CID.from(cid).value);
 
-  verbose && log(`Registering ${resource}.` + (record.tag ? ` Tagged ${record.tag.join(', ')}.` : '') + (record.version ? ` Version ${record.version}.` : ''));
+  verbose && log(`Registering ${resource}.` + (record.tag ? ` Tagged ${record.tag.join(', ')}.` : ''));
 
   if (verbose || noop) {
     log(JSON.stringify({ record }, undefined, 2));
   }
 
-  const client: { registryClient: IRegistryClient } = await getDXNSClient();
+  const client: { registryClient: RegistryClient } = await getDXNSClient();
 
-  const recordType = await client.registryClient.getResourceRecord(DXN.parse(type), 'latest');
-  assert(recordType);
-  assert(recordType.record.kind === RecordKind.Type, `Can not find type "${type}" in DXNS.`);
+  const typeResource = await client.registryClient.getResource(DXN.parse(type));
+  assert(typeResource, `Can not find type "${type}".`);
+  const typeCid = typeResource.tags.latest;
+  assert(typeCid, `No latest tag registered for type "${type}".`);
 
   let recordCID;
   if (!noop) {
-    recordCID = await client.registryClient.insertDataRecord(record, recordType?.record.cid, {
+    recordCID = await client.registryClient.registerRecord(record, typeCid, {
       description,
       displayName,
       tags
     });
   }
 
-  const domainKey = await client.registryClient.resolveDomainName(domain);
-  const opts: UpdateResourceOptions = { version: record.version, tags: record.tag ?? ['latest'] };
```

Comment:
We need to make sure publishing latest apps/bots from CI is still happy with that (I guess atm `--skipExisting` together with `--version` helps with release-please)